### ISSUE TITLE
"Common Language Runtime detected an invalid program." after upgrade to Datadog.Trace 1.14

### ISSUE THREAD

**Issue by d47id** [DataDog/dd-trace-dotnet#762]
**Describe the bug**
Datadog.Trace versions >1.7 and <1.14 did not successfully trace Dapper/Npgsql calls. Since we upgraded from 1.7 all the way to 1.14, we've gotten sporadic "Common Language Runtime detected an invalid program" errors. We recently upgraded to 1.17 and there has been no change.

**To Reproduce**
I don't have reliable steps to reproduce. Sometimes a pod will start up and any attempt to access Postgres via Npgsql will throw this error. Restarting the k8s deployment fixes the issue in all cases. 

**Expected behavior**
This error does not occur.

**Screenshots**
All the recent examples in our logs look like this:
```Common Language Runtime detected an invalid program.
at Npgsql.PostgresDatabaseInfo.LoadBackendTypes(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Npgsql.PostgresDatabaseInfo.LoadBackendTypes(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at Npgsql.PostgresDatabaseInfo.LoadPostgresInfo(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at Npgsql.PostgresDatabaseInfoFactory.Load(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at Npgsql.NpgsqlDatabaseInfo.Load(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
   at Npgsql.NpgsqlConnector.LoadDatabaseInfo(NpgsqlTimeout timeout, Boolean async)
   at Npgsql.NpgsqlConnector.Open(NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
   at Npgsql.ConnectorPool.AllocateLong(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlConnection.<>c__DisplayClass32_0.<<Open>g__OpenLong|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Npgsql.NpgsqlConnection.Open()
```

**Runtime environment (please complete the following information):**
- Instrumentation mode: Npgsql/Dapper instrumentation is automatic. We have manual instrumentation for gRPC via a custom-built interceptor. 
- Tracer version: 1.17.0
- OS: linux containers based on mcr.microsoft.com/dotnet/core/aspnet:3.1.2-buster-slim running on GKE
- CLR: .Net Core 3.1.2

**Comment by lucaspimentel** [comment]
@d47id: thanks for the reporting and for the stack trace. This issue is assigned to the current sprint so we will be look into it soon.

For reference, it looks like we've had similar errors reported in the past: #564 and #568.

**Comment by hendonnelly** [comment]
Hi @lucaspimentel,
Just wondering if there is any update on this issue? We are experiencing it too.

Thanks!

**Comment by kevingosse** [comment]
A fix for InvalidProgramException was released in 1.19.2. Feel free to open a new issue if you run into this problem again.
